src/mongoose.ts

```diff
@@ -111,89 +111,60 @@ export function htMongooseFactory(mongoose: any) {
     };
   }
 
-  function WithSaveOnDocument(docKey: string) {
-    // tslint:disable-next-line:only-arrow-functions
-    return function<TSuper extends Constructor>(Super: TSuper) {
-      return class SaveOnDocument extends Super {
-        constructor(...args: any[]) {
-          super(...args);
-        }
-        public async doWork() {
-          if (Object.keys(this).includes(docKey)) {
-            try {
-              await (this as any)[docKey].save();
-            } catch (err) {
-              throw Boom.badData(
-                'Unable to save. Please check if data sent was valid.'
-              );
-            }
-          } else {
-            throw Boom.badRequest('Resource not found');
-          }
-        }
-      };
-    };
+  function WithSaveOnDocument(entityToCreate: string) {
+    return WithDoWork(async requestData => {
+      if (Object.keys(requestData).includes(entityToCreate)) {
+        return await requestData[entityToCreate].save();
+      } else {
+        throw Boom.badRequest('Resource not found');
+      }
+    });
   }
 
-  function WithUpdateByBody(entityToUpdate: string) {
-    // tslint:disable-next-line:only-arrow-functions
-    return function<TSuper extends Constructor>(Super: TSuper) {
-      return class WithUpdatingBody extends Super {
-        constructor(...args: any[]) {
-          super(...args);
-        }
-        public async doWork() {
-          if (Object.keys(this as any).includes(entityToUpdate)) {
-            // tslint:disable:no-string-literal
-            (this as any)[entityToUpdate].set((this as any).body);
-            await (this as any)[entityToUpdate].save();
-          } else {
-            throw Boom.badRequest('Resource not found');
-          }
-        }
-      };
-    };
+  function WithUpdateByBody(entityToUpdate: string, updateWithEntity: string) {
+    return WithDoWork(async requestData => {
+      if (Object.keys(requestData).includes(entityToUpdate)) {
+        await requestData[entityToUpdate].set(requestData[updateWithEntity]);
+        return await requestData[entityToUpdate].save();
+      } else {
+        throw Boom.badRequest('Resource not found');
+      }
+    });
   }
```

Comment:
as mentioned in STAF-2255, the above shouldn't save() - we already have a save.  The customer should just call save separately.

Also please change the names
1. function name to WithUpdateDocument
2. updateWithEntity -> propertyKeyWithNewData
3. entityToUpdate->propertyKeyOfDocument

Reply:
done